test/integration/react-streaming-and-server-components/test/rsc.js

```diff
@@ -30,6 +30,18 @@ export default function (context, { runtime, env }) {
     expect(homeHTML).toContain('foo.client')
   })
 
+  it('should reuse the inline flight response without sending extra requests', async () => {
+    let hasFlightRequest = false
+    const browser = await webdriver(context.appPort, '/')
+    browser.on('request', (request) => {
+      const url = request.url()
+      if (/__flight__=1/.test(url)) {
+        hasFlightRequest = true
+      }
+    })
+    expect(hasFlightRequest).toBe(false)
```

Comment:
Should we wait for a couple of seconds before checking and exiting?

Reply:
Added a new handler to collect all requests starting from page load. By default webdriver waits for hydration so we can cover it with that handler.

Reply:
If it waits for hydration, adding the listener `browser.on('request', (request) => {` after that will not make any difference I think?

Reply:
This new change looks good to me.